reverse_proxy_handler.cc

```diff
@@ -0,0 +1,142 @@
+#include <string>
+#include <memory>
+
+#include "request_handler.h"
+#include "reverse_proxy_handler.h"
+#include "response.h"
+#include "config_parser.h"
+#include "http_client.h"
+
+#include <algorithm>
+
+#include <boost/log/trivial.hpp>
+
+RequestHandler::Status ReverseProxyHandler::Init(const std::string& uri_prefix, const NginxConfig& config)
+{
+	prefix_ = uri_prefix;
+
+	for (auto statement : config.statements_)
+	{
+		if(statement->tokens_.size() == 2 && statement->tokens_[0] == "proxy_pass")
+		{
+			url_ = statement->tokens_[1];
+
+			std::string::size_type protocol_pos = url_.find("//");
+			if(protocol_pos == std::string::npos) {
+				BOOST_LOG_TRIVIAL(error) << "proxy_pass didn't specify protocol.'";
+				return RequestHandler::ERROR;
+			}
+			protocol_ = url_.substr(0, protocol_pos - 1);
+
+			std::string::size_type host_pos = url_.find('/', protocol_pos + 2);
+			if(host_pos != std::string::npos) {
+				host_ = url_.substr(protocol_pos + 2, host_pos - protocol_pos - 2);
+				path_ = url_.substr(host_pos);
+			}
+			else {
+				host_ = url_.substr(protocol_pos + 2);
+				path_ = "/";
+			}
+			printf("path_: %s\n", path_.c_str());
+			return RequestHandler::OK;
```

Comment:
What do you mean specifying a port should be supported? Can you give an example of a config?

And yeah I didn't mean to log it at all; was a debug line.